bims/utils/user_permissions.py

```diff
@@ -0,0 +1,19 @@
+# coding=utf-8
+from braces.views import LoginRequiredMixin
+from django.http import Http404
+
+
+class ValidatorRequiredMixin(LoginRequiredMixin):
+
+    def dispatch(self, request, *args, **kwargs):
+        validator = \
+            'validator' in request.user.groups.all().values_list(
+                'name', flat=True)
+
+        if not request.user.is_authenticated():
+            return self.handle_no_permission(request)
+        elif not validator:
+            raise Http404
```

Comment:
Please don't return this, use similar failed result from LoginRequiredMixin.

Reply:
The return from LoginRequiredMixin when user is not login is redirect to login page then coming back to this page. Should the user re login when user is not a validator?

Reply:
yes, I think it's like when you're not an admin but you go to admin page. It will go to login page.